PROTOCOL.md

```diff
@@ -2801,6 +2810,107 @@ Expression | Example
+
+The server may try its best to filter files based on the predicates in a **BEST EFFORT** mode. The server may return all the files if validations fail. So clients should apply these predicates to the filter the data returned by the server.
+
+We will deprecate the legacy predicateHints once all the delta sharing client and servers have moved to the json predicate based file filtering.
+
+The JSON format:
+
+Field Name | Descrption
+-|-
+op | The name of the operation. See the section on supported ops for details.
+children | The child operations for the op. This allows us to represent a predicate tree. Leaf ops do not have any children. Unary ops such as **not** have only one child. Binary ops such as **equal** have two children. Nary ops such as **and** can have multiple children.
+name | Specifies the name of a column. This is only applicable to column ops.
+value | Specifies the value of a literal. This is only applicable to literal ops.
+valueType | Specifies the value type of a column or a literal op. This is only applicate to column and literal ops.
+
+The supported Ops:
+
+Op | Op Type | Description
+-|-|-
+column | Leaf | Represents a column. A column op has two fields: **name** and **valueType**. The column's value will be cast to the specified **valueType** during comparisons.
+literal | Leaf | Represents a literal or fixed value. A literal op has two fields: **value** and **valueType**. The literal's **value**  will be cast to the specified **valueType** during comparisons.
+isNull | Unary | Represents a null check on a column op. This op should have only one child, the column op.
+equal | Binary | Represents an equality ("=") check. This op should have two children, and both should be leaf ops.
+lessThan | Binary | Represents a less than ("<") check. This op should have two children, and both should be leaf ops.
+lessThanOrEqual | Binary | Represents a less than or equal ("<=") check. This op should have two children, and both should be leaf ops.
+greaterThan | Binary | Represents a greater than (">") check. This op should have two children, and both should be leaf ops.
+greaterThanOrEqual | Binary | Represents a greater than (">=") check. This op should have two children, and both should be leaf ops.
+and | Nary | Represents a logical and operation amongst its children. This op should have at least two children.
+or | Nary | Represents a logical or operation amongst its children. This op should have at least two children.
+not | Unary | Represents a logical not check. This op should have once child.
+
+The supported value types:
+
+ValueType | Description
+-|-
+"bool" | Represents an Boolean type.
+"int" | Represents an Integer type.
+"long" | Represents a Long type.
+"string" | Represents a String type.
+"date" | Represents a Date type.
+
+
+Examples
+  - Note that the json should be encoded using Base64 encoding.
```

Comment:
Could you point me to the code where this encoding is applied?

Reply:
https://src.dev.databricks.com/delta-io/delta-sharing/-/blob/spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala?L119:9&subtree=true#tab=references